business/checkers/authorization/no_host_checker.go

```diff
@@ -80,21 +83,13 @@ func (n NoHostChecker) validateHost(ruleIdx int, to []*api_security_v1beta.Rule_
 
 func (n NoHostChecker) hasMatchingService(host kubernetes.Host, itemNamespace string) bool {
 	// Covering 'servicename.namespace' host format scenario
-	localSvc, localNs := kubernetes.ParseTwoPartHost(host)
+	_, localNs := kubernetes.ParseTwoPartHost(host)
 
 	// Check wildcard hosts - needs to match "*" and "*.suffix" also..
 	if host.IsWildcard() && localNs == itemNamespace {
 		return true
 	}
 
-	// Only find matches for workloads and services in the same namespace
-	if localNs == itemNamespace {
-		// Check ServiceNames
-		if matches := n.ServiceList.HasMatchingServices(localSvc); matches {
-			return matches
-		}
-	}
-
 	// Check ServiceEntries
 	if kubernetes.HasMatchingServiceEntries(host.String(), n.ServiceEntries) {
```

Comment:
Already commented, that ServiceEntries would be included in the ServiceList fetched from the Istio Registry.
So, it doesn't sound necessary, but it can be improved in the next iteration.